pkg/credentialprovider/azure_credentials.go

```diff
@@ -224,10 +225,16 @@ func (a *acrProvider) getFromACR(ctx context.Context, loginServer string) (strin
 // If the provided image is not an acr image, this function will return an empty string.
 func (a *acrProvider) parseACRLoginServerFromImage(image string) (string, string) {
 	targetImage, sourceRegistry := a.processImageWithRegistryMirror(image)
-
-	match := acrRE.FindAllString(targetImage, -1)
-	if len(match) == 1 {
-		targetRegistry := match[0]
+	targetRegistry := acrRE.FindString(targetImage)
+	imageWithoutRegistry := strings.Trim(targetImage, targetRegistry)
+	// for non customer cloud case, return registry only when:
+	//   - the acr pattern match
+	//   - the left string is empty or // credential provider authenticates the image pull request, but not validates the existence of the image
```

Comment:
fortunately, use acr with repo will raise authorization failure because kubelet will pull this kind of image from docker hub

Failed to pull image "qingchuanhao.azurecr.io": failed to pull and unpack image "docker.io/library/qingchuanhao.azurecr.io:latest": failed to resolve reference "docker.io/library/qingchuanhao.azurecr.io:latest": pull access denied, repository does not exist or may require authorization: server message: insufficient_scope: authorization failed

Reply:
```
  containers:
  - name: container
    image: qingchuanhao.azurecr.io
```